### PR TITLE
feat: use more efficient DirectX overlay

### PR DESCRIPTION
**Pull request by Xenthalon**
The game was visibly chugging with the map up, and I traced it to the Windows.Forms Refresh() call. But even changing it to Invalidate(), and restricting it's calls to happen less often did not aleviate the problem. Windows.Forms is just not made for game overlays.

So initially I wanted to just use SharpDX to draw the map more efficiently with DirectX, but stumbled over Gameoverlay.NET, which also uses SharpDX under the hood but helps a lot with Overlay positioning over another process, helping us out a lot. This should get rid of a lot of the issues about overlay placement with multiple monitors and such.

I haven't quite figured out the math in Overlay mode for TopLeft and TopRight alignment, I could use some help there, but Overlay with Center works fine, and the other placements as well.

I had to throw in a TrayIcon last minute, because without a Windows.Forms window the taskbar icon was gone, so that part looks a bit messy.

tl;dr: this switches the overlay renderer to use DirectX, chugging is gone and the overlay stays inside the game process window at all times

### CODE REVIEW

**Comment by admon84** [comment]
The overlay rendering performance does feel improved with this change.  The system tray icon takes a little getting used to.

My only complaint is that exceptions appear to crash MapAssist **silently**.  For example if you forget to start the map API, MapAssist will close (tray icon only disappears when you mouse-over it) and no exception messages are displayed.  Can that be fixed?

**Comment by Xenthalon** [comment]
Good catch, I added a global exception handler.
